src/gc-pages.c

```diff
@@ -33,9 +32,9 @@ void jl_gc_init_page(void)
 
 // Try to allocate a memory block for multiple pages
 // Return `NULL` if allocation failed. Result is aligned to `GC_PAGE_SZ`.
-static char *jl_gc_try_alloc_pages(int pg_cnt) JL_NOTSAFEPOINT
+char *jl_gc_try_alloc_pages(void) JL_NOTSAFEPOINT
 {
-    size_t pages_sz = GC_PAGE_SZ * pg_cnt;
+    size_t pages_sz = GC_PAGE_SZ * block_pg_cnt;
```

Comment:
This is optimistically too large, which is why we had the exponential backoff logic, to deal with constrained virtual spaces such as 32-bit or with ulimit. It does not seem like we need to delete that logic, which maybe a regression for some users who are near their fixed limits.